### PR TITLE
Added request to extern api open library

### PR DESCRIPTION
**Pull request by blacksam07**
# Summary
- Added [httparty](https://github.com/jnunemaker/httparty) gem
- Added request to search by ISBN in openlibrary

# Trello Cards
[Integración api externa](https://trello.com/c/u0sjAg77/35-2-integraci%C3%B3n-api-externa)

# Screenshots
image

### CODE REVIEW

File: app/controllers/api/v1/open_library_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module Api
+  module V1
+    class OpenLibraryController < ApiController
+      def show
+        books = OpenLibrary.new(params[:isbn]).books
+        render json: books
```
Comment (mcolombo87) [inline app/controllers/api/v1/open_library_controller.rb]: What happens if the service does not respond with a 200 or does not respond at all (timeout)?

File: app/services/open_library.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    @options = { query: { bibkeys: "ISBN:#{isbn}", format: 'json', jscmd: 'data' } }
+  end
+
+  def books
+    process_response self.class.get('/api/books', @options)
+  end
+
+  def process_response(response)
+    result = { ISBN: @isbn }
+    response_body = JSON.parse(response.body)
```
Comment (mcolombo87) [inline app/services/open_library.rb]: Only for general knowledge! (You not need change it) You can resolve this just like that...
```ruby
response_body = JSON.parse(response.body)
keys = %w[title subtitle number_of_pages authors]
keys.each do |key|
  result[key.to_sym] = response_body.dig(@isbn, key)
end
result
```
It's better? Not? Maybe? Yes? I don't know, it just I'm bored. 😄

Comment (holywyvern) [reply app/services/open_library.rb]: I think a better alternative is:

```rb
response_body =  JSON.parse(response.body)
response_body[@isbn]
  .slice(:title, :subtitle, :number_of_pages, :authors)
  .merge(ISBN: @isbn) 
```

But it doesn't make a lot of difference

File: app/services/open_library.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  base_uri 'https://openlibrary.org'
+
+  def initialize(isbn)
+    @isbn = "ISBN:#{isbn}"
+    @options = { query: { bibkeys: "ISBN:#{isbn}", format: 'json', jscmd: 'data' } }
+  end
+
+  def books
+    response = self.class.get('/api/books', @options)
+    raise 'External error' if response.code != 200
```
Comment (mcolombo87) [inline app/services/open_library.rb]: I thinks you may do a 
```ruby
raise 'External error' if response.ok?
```
Instance of that... but it's fine.

Comment (blacksam07) [reply app/services/open_library.rb]: i think a best way

```suggestion
    raise 'External error' unless response.ok?
```

Comment (mcolombo87) [reply app/services/open_library.rb]: Yes, that is it! :+1: :+1:

**Pull request merged**